src/redis-cli.c

```diff
@@ -418,91 +425,226 @@ static sds cliVersion(void) {
+                }
+                else if (!strcmp(flag, "multiple_token")) {
+                    multipleToken = 1;
+                }
+            }
+        }
+    }
+
+    // Then build the "repeating part" of the argument string.
+    if (!strcmp(type, "key") ||
+        !strcmp(type, "string") ||
+        !strcmp(type, "integer") ||
+        !strcmp(type, "double") ||
+        !strcmp(type, "pattern") ||
+        !strcmp(type, "unix-time") ||
+        !strcmp(type, "token")) {
+            repeatPart = sdscat_orempty(repeatPart, name);
+    }
+    else if (!strcmp(type, "oneof")) {
+        repeatPart = cliConcatArguments(repeatPart, arguments, "|");
+    }
+    else if (!strcmp(type, "block")) {
+        repeatPart = cliConcatArguments(repeatPart, arguments, " ");
+    }
+    else if (strcmp(type, "pure-token") != 0) {
+        fprintf(stderr, "Unknown type '%s' set for argument '%s'\n", type, name);
+    }
+
+    // Finally, build the parameter string.
+    if (tokenPart[0] != '\0' && strcmp(type, "pure-token") != 0) {
+        tokenPart = sdscat(tokenPart, " ");
+    }
+    if (optional) {
+        params = sdscat(params, "[");
+    }
+    params = sdscat(params, tokenPart);
+    params = sdscat(params, repeatPart);
+    if (multiple) {
+        params = sdscat(params, " [");
+        if (multipleToken) {
+            params = sdscat(params, tokenPart);
+        }
+        params = sdscat(params, repeatPart);
+        params = sdscat(params, " ...]");
+    }
+    if (optional) {
+        params = sdscat(params, "]");
+    }
+    sdsfree(tokenPart);
+    sdsfree(repeatPart);
+    return params;
+}
+
+/* Initialize a command help entry for the command/subcommand described in reply.
+ * Returns the index of the next available position in the help entries table. */
+static int cliInitCommandHelpEntry(char *cmdname, char *subcommandname, int helpIndex, redisReply *reply) {
+    helpEntry *help = &(helpEntries[helpIndex++]);
+    help->argc = subcommandname ? 2 : 1;
+    help->argv = zmalloc(sizeof(sds) * help->argc);
+    help->argv[0] = sdsnew(cmdname);
+    sdstoupper(help->argv[0]);
+    if (subcommandname) {
+        help->argv[1] = sdsnew(subcommandname);
+        sdstoupper(help->argv[1]);
+    }
+    sds fullname = sdsnew(help->argv[0]);
+    if (subcommandname) {
+        fullname = sdscat(fullname, " ");
+        fullname = sdscat(fullname, help->argv[1]);
+    }
+    help->full = fullname;
+    help->type = CLI_HELP_COMMAND;
+
+    struct commandHelp *ch = zmalloc(sizeof(*ch));
+    help->org = ch;
+    ch->name = help->full;
+    ch->params = sdsempty();
+
+    if (reply->elements > 7) {
+        redisReply *map = reply->element[7];
+        for (size_t j = 0; j < map->elements; j += 2) {
+            char *key = map->element[j]->str;
+            if (!strcmp(key, "summary")) {
+                redisReply *reply = map->element[j + 1];
+                ch->summary = sdsnew(reply->str);
+            }
```

Comment:
FYI #10028

Reply:
So I'll put this on hold until that PR is finalised.